includes/class-newspack-popups-inserter.php

```diff
@@ -658,23 +658,29 @@ public static function enqueue_scripts() {
 				true
 			);
 
-			$segments = Newspack_Popups_Segmentation::get_segments( false );
-
-			// Gather segments for all prompts to be displayed.
-			foreach ( $segments as $segment ) {
-				if ( ! empty( $segment ) && ! empty( $segment['criteria'] ) && ! isset( self::$segments[ $segment['id'] ] ) ) {
-					self::$segments[ $segment['id'] ] = [
-						'criteria' => $segment['criteria'],
-						'priority' => $segment['priority'],
-					];
-				}
-			}
-
 			$script_data = [
-				'debug'    => defined( 'WP_DEBUG' ) && WP_DEBUG,
-				'segments' => self::$segments,
+				'debug' => defined( 'WP_DEBUG' ) && WP_DEBUG,
 			];
 
+			if ( class_exists( '\Newspack\Reader_Data' ) ) {
+				$segments = Newspack_Popups_Segmentation::get_segments( false );
+
+				// Gather segments for all prompts to be displayed.
+				foreach ( $segments as $segment ) {
+					if ( ! empty( $segment ) && ! empty( $segment['criteria'] ) && ! isset( self::$segments[ $segment['id'] ] ) ) {
+						self::$segments[ $segment['id'] ] = [
+							'criteria' => $segment['criteria'],
+							'priority' => $segment['priority'],
+						];
+					}
+				}
+
+				$script_data['segments'] = self::$segments;
+
+			} else {
+				$script_data['segmentation_disabled'] = true;
```

Comment:
nit: can we avoid the negative name here? It can become very confusing (as we saw yesterday)

Reply:
Good call—d4b9a25 removes it, as there's really no need for a separate variable at all. More simply, if there are no `segments` to handle, then there's no need to use the reader data library at all.

Reply:
Whoops, ef8ac67 fixes a bug in the prior commit which assumed that `newspack_popups_view.segments` would be an array and therefore have a `length` property, when in fact it's a keyed object.